core/src/main/java/io/kestra/core/runners/RunVariables.java

```diff
@@ -0,0 +1,301 @@
+package io.kestra.core.runners;
+
+import com.google.common.collect.ImmutableMap;
+import io.kestra.core.models.executions.Execution;
+import io.kestra.core.models.executions.TaskRun;
+import io.kestra.core.models.flows.Flow;
+import io.kestra.core.models.flows.Input;
+import io.kestra.core.models.flows.input.SecretInput;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.models.tasks.common.EncryptedString;
+import io.kestra.core.models.triggers.AbstractTrigger;
+import io.kestra.core.models.triggers.Trigger;
+import io.kestra.core.secret.Secret;
+import lombok.AllArgsConstructor;
+import lombok.With;
+
+import java.security.GeneralSecurityException;
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Class for building {@link RunContext} variables.
+ */
+public abstract class RunVariables {
```

Comment:
I prefer to use final class with a private constructor as abstract denote inheritance.